src/MomentoTaggedCache.php

```diff
@@ -49,9 +48,30 @@ public function get($key, $default = null): mixed
     /**
      * @throws UnknownError
      */
-    public function flush()
+    public function flush(): bool
     {
-        throw new UnknownError("flush operations is currently not supported.");
+        $tags = $this->tags->getNames();
+        if (!self::validateTags($tags)) {
+            throw new UnknownError("Empty tag is not valid. Please provide non-empty tags to flush.");
+        }
+        foreach ($tags as $tag) {
+            $keys = $this->store->setFetch($tag);
+            if (!$keys) {
+                return false;
```

Comment:
I'm a little confused as to what to expect this method to do - looks like if there is any empty set for any reason we just bail instead of continuing to delete the rest of the data? It leaves partial deletes as-is too, so it seems. Is this the right thing to do?